dotnet/src/Connectors/Connectors.Memory.CosmosDB/CosmosDBMemoryStore.cs

```diff
@@ -95,64 +99,53 @@ public async Task CreateCollectionAsync(string collectionName, CancellationToken
     /// <inheritdoc />
     public Task<bool> DoesCollectionExistAsync(string collectionName, CancellationToken cancel = default)
     {
-        var container = this._database.Client.GetContainer(this._databaseName, collectionName);
-        return Task.FromResult(container != null);
+        // Azure Cosmos DB does not support checking if container exists without attempting to create it. This does not break the interface but it is not ideal.
```

Comment:
Is creating containers idempotent? If so, that would be good to mention here.

Reply:
Could you do a container get? https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container.deletecontainerasync?view=azure-dotnet

Reply:
unfortunately no :( the behavior is mentioned in the comment